### PR TITLE
Add splash screen

### PR DESCRIPTION
**Pull request by andrelegault**
## Description
This pull request adds a splash screen to the app that pre-loads shared data (`.json` files).

Currently, we have static function in each of the model/service requiring data. These static functions are called (as shown below) while the splash screen is shown. When the static functions (futures) all complete, the splash screen disappears.

The function preloading all the data is found in: https://github.com/MichelRahme/ConcordiaNavigation/blob/58d9e69379c9e41c2ca17b5d3147d555f10368a5/lib/screens/splash_screen.dart#L22-L29

and the invocation is found in: https://github.com/MichelRahme/ConcordiaNavigation/blob/58d9e69379c9e41c2ca17b5d3147d555f10368a5/lib/screens/splash_screen.dart#L44

## Related Issues
#156

### CODE REVIEW

File: lib/services/change_later.dart [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+            floorNumber: int.parse(
+              data['buildings'][i]['floors'][j]['number'],
+            ),
+            classrooms: classroomSet,
+          ),
+        );
+        buildingSet.add(
+          Building(
+            buildingInitial: data['buildings'][i]['initial'],
+            floors: floorSet,
```
Comment (MichelRahme) [inline lib/services/change_later.dart:112]: Why `floors` set is not in the Building Object ? It should be there

**Comment by andrelegault** [comment]
I think dev is messed up

File: test/providers/shuttle_service_test.dart [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+      expect(data, "Next Shuttle Bus at: 07:45");
+    });
+
+    test(
+        'getNextShuttle() returns string with next sgw shuttle at 11:01pm Monday',
+        () async {
+      DateTime time = new DateTime(2020, 1, 6, 23, 01);
+      String data = ShuttleService.getNextShuttle("sgw", time);
+      expect(data, "Next Shuttle Bus at: 07:45");
+    });
```
Comment (EvanMateo7) [inline test/providers/shuttle_service_test.dart:61]: The tests are fine. 
Thanks for working on them.

File: lib/screens/splash_screen.dart [hunk 3]
```diff
@@ -41,11 +40,11 @@ class _SplashScreenState extends State<SplashScreen>
...
-    loadAssets()
-        .then((value) => Navigator.pushReplacementNamed(context, '/home'));
-    // Timer(Duration(seconds: 3), () {
-    //   Navigator.pushNamed(context, '/home');
-    // });
+    loadAssets().then((value) {
+      Timer(Duration(seconds: 2), () {
+        Navigator.pushReplacementNamed(context, '/home');
+      });
+    });
```
Comment (EvanMateo7) [inline lib/screens/splash_screen.dart:45]: Why is there a timer here?
I'm guessing its for a smoother animation if loadAssets() is too fast?

Comment (MichelRahme) [reply lib/screens/splash_screen.dart:45]: Yes exactly this.

**Pull request merged**